arm9/src/unk_02006D98.c

```diff
@@ -0,0 +1,1747 @@
+    ptr->unk298 = heap_id;
+    ptr->unk2E0 = 0;
+    ptr->unk29C = 0;
+    ptr->unk2A0 = 0x8000;
+    ptr->unk2A4 = 0;
+    ptr->unk2A8 = 0x80;
+
+    ptr->unk2AC = AllocFromHeap(heap_id, 0x8000);
+
+    ptr->unk2B0 = AllocFromHeap(heap_id, 0xc0);
+    MIi_CpuClearFast(0, ptr->unk2B0, 4);
+
+    ptr->unk2B4 = AllocFromHeap(heap_id, 0xc0);
+    MIi_CpuClearFast(0, ptr->unk2B4, 4);
+
+    for (int i = 0; i < 4; i++)
+    {
+        MIi_CpuClearFast(0, &ptr->unk000[i], sizeof(struct UnkStruct_02006D98_2));
+    }
+
+    NNS_G2dSetupSoftwareSpriteCamera();
+    ptr->unk2E3 = 0;
+
+    u8 *st8;
+    void *st4 = AllocAndReadWholeNarcMemberByIdPair(
+        NARC_POKETOOL_POKEGRA_OTHERPOKE, NARC_otherpoke_narc_0211_NCGR, ptr->unk298);
+    NNSG2dCharacterData *stc;
+    NNS_G2dGetUnpackedCharacterData(st4, &stc);
+
+    ptr->unk2B8.pixelFmt = stc->pixelFmt;
+    ptr->unk2B8.mapingType = stc->mapingType;
+    ptr->unk2B8.characterFmt = stc->characterFmt;
+    st8 = stc->pRawData;
+
+    FUN_02008A74(st8);
+    MI_CpuFill8(ptr->unk2AC, *st8, 0x8000);
+
+    for (int i = 0; i < 0x50; i++)
+    {
+        for (int j = 0; j < 0x28; j++)
+        {
+            ((u8 *)ptr->unk2AC)[0x5050 + i * 0x80 + j] = st8[i * 0x50 + j];
+        }
+    }
+
+    FreeToHeap(st4);
+    ptr->unk2E1 = 1;
+    ptr->unk2E2 = 1;
+
+    return ptr;
+}
+
+THUMB_FUNC void FUN_02006ED4(struct UnkStruct_02006D98 *param0)
+{
+    s32 arg3;
+    s32 arg4;
+
+    FUN_020082A8(param0);
+    FUN_020086F4(param0);
+
+    NNS_G3dGeFlushBuffer();
+
+    reg_G3_MTX_PUSH = 0;
+
+    G3_TexImageParam(param0->unk260.attr.fmt,
+        GX_TEXGEN_TEXCOORD,
+        param0->unk260.attr.sizeS,
+        param0->unk260.attr.sizeT,
+        GX_TEXREPEAT_NONE,
+        GX_TEXFLIP_NONE,
+        param0->unk260.attr.plttUse,
+        param0->unk29C);
+
+    for (int st18 = 0; st18 < 4; st18++)
+    {
+        if (param0->unk000[st18].unk00_0 == 0 || param0->unk000[st18].unk54_0 != 0 ||
+            param0->unk000[st18].unk54_B != 0)
+        {
+            continue;
+        }
+
+        if (param0->unk000[st18].unk68 != NULL)
+        {
+            param0->unk000[st18].unk68(&param0->unk000[st18], &param0->unk000[st18].unk24);
+        }
+
+        NNS_G3dGeFlushBuffer();
+
+        if (param0->unk2E3 != 1)
+        {
+            reg_G3_MTX_IDENTITY = 0;
+        }
+
+        FUN_02007F48(&param0->unk000[st18]);
+
+        u32 shift;
+        if (param0->unk260.attr.fmt == 2)
+        {
+            shift = 1;
+        }
+        else
+        {
+            shift = 0;
+        }
+
+        reg_G3_TEXPLTT_BASE = (param0->unk2A4 + st18 * 0x20) >> (4 - shift);
+
+        u32 r1 = param0->unk000[st18].unk28 << 0xc;
+        u32 r0 = (u32)((param0->unk000[st18].unk26 + param0->unk000[st18].unk42) << 0xc);
+        u32 r3 = (u32)((param0->unk000[st18].unk24 + param0->unk000[st18].unk40) << 0xc);
+
+        reg_G3_MTX_TRANS = r3;
+        reg_G3_MTX_TRANS = r0;
+        reg_G3_MTX_TRANS = r1;
```

Comment:
register access usually happens within the scope of an sdk inline

Reply:
I added all the inlines used in this file, it matched without any effort